### PR TITLE
Implement prometheus exporter provider

### PR DESCRIPTION
**Pull request by jack-berg**
Related to #4949.

Prometheus is implemented as a `MetricReader` rather than `MetricExporter` so implementing `ConfigurableMetricExporterProvider` isn't an option. This PR takes a different approach and implements the `AutoConfigurationCustomizerProvider`, which allows `PrometheusCustomizerProvider` to customize the `SdkMeterProviderBuilder` with `PrometheusHttpServer` when `otel.metrics.exporter=prometheus`. 

We could also add a new SPI for providing named `MetricReader`s, but we don't actually support custom implementations of `MetricReader` right now so I don't think that's a good idea.

### CODE REVIEW

File: sdk-extensions/autoconfigure/build.gradle.kts [hunk 1]
```diff
@@ -14,8 +14,6 @@ dependencies {
 
   implementation(project(":semconv"))
 
-  compileOnly(project(":exporters:prometheus"))
```
Comment (jack-berg) [inline sdk-extensions/autoconfigure/build.gradle.kts:17]: Last exporter compile dependency is gone! Woot!

**Review by jkwatson — APPROVED** [review]

File: sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ClasspathUtil.java [hunk 2]
```diff
@@ -9,11 +9,18 @@
 
 final class ClasspathUtil {
 
-  @SuppressWarnings("UnusedException")
-  static void checkClassExists(String className, String featureName, String requiredLibrary) {
+  static boolean checkClassExists(String className) {
```
Comment (mateuszrzeszutek) [inline sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ClasspathUtil.java:12]: Nit: both of these methods are named the same while being completely different in what they do - one of them does not throw and you have to check the return value, the other one does not return anything and the caller has to rely on an exception being thrown.
How about renaming this one to `classExists` or `doesClassExist`?

**Review by mateuszrzeszutek — APPROVED** [review]

File: sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ClasspathUtil.java (hunk 2 above)
Comment (jack-berg) [reply sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ClasspathUtil.java:12]: Good point. I like `classExists`. The `check` prefix is associated with methods that throw if some condition is not met, i.e. `Preconditions.checkArgument`.

Comment (jack-berg) [reply sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ClasspathUtil.java]: Turns out that there's just a single use of `ClasspathUtil` left and its a call to `ClasspathUtil.classExists`. I've moved the `try / catch` logic to the calling location and deleted `ClaspathUtil`.

**Pull request merged**